python/test/unit/common/test_index_map.py

```
@@ -10,54 +10,55 @@
 
-    # Create sub-index map, and map from ghost posittion in new map to
+    # Create sub index map and a map from the ghost position in new map to the
     # position in old map
-    submap, ghosts_pos_sub = map.create_submap(local_indices)
+    submap, ghosts_pos_sub = map.create_submap(local_indices[my_rank])
 
     # Check local and global sizes
-    assert submap.size_local == local_size_sub
+    assert submap.size_local == submap_local_size[my_rank]
     assert submap.size_global == sum([rank + rank % 2 for rank in range(comm.size)])
 
-    # Running this code on 2 procs gives you that the ghost on process 0
-    # is owned by process 0
-    # print(comm.rank, "Range", map.local_range, "Local", local_indices,
-    #       "Sub range", submap.local_range, "Sub ghosts", submap.ghosts,
-    #       "Ghosts", map.ghosts, "Sub owners", owners_sub)
-    assert (dest_ranks == owners).all()
-
     # Check that first rank has no elements
     if comm.rank == 0:
         assert submap.size_local == 0
 
     # Check that rank on sub-process ghosts is the same as the parent
     # map
+    owners = map.ghost_owner_rank()
+    assert (dest_ranks == owners).all()
     assert (owners[ghosts_pos_sub] == submap.ghost_owner_rank()).all()
 
-    # print(f"{myrank}: , {submap.ghosts}, {owners_sub},\
-    #       {map.ghosts[ghosts_pos_sub]} {np.array(owners, dtype=np.int32)[ghosts_pos_sub]}")
+    # Check that ghost indices are correct in submap
+    # NOTE This assumes size_local is the same for all ranks
+    # TODO Consider renaming to something shorter
+    submap_global_to_map_global_map = np.concatenate([local_indices[rank] + map_local_size * rank
+                                                      for rank in range(comm.size)])
+    # FIXME Do this more elegantly
+    submap_ghosts = []
+    for map_ghost in map.ghosts:
+        submap_ghost = np.where(submap_global_to_map_global_map == map_ghost)[0]
+        if submap_ghost.size != 0:
```

If we cannot find the ghost I guess we should raise an error?

---

You mean raise an error if `submap_ghost.size == 0`? This shouldn't raise an error, as a ghost in the original map might not be in the submap